src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
                         typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        "FileProvider"));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
+            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(fileProvider.GetFileInfo("/").PhysicalPath));
```

Comment:
I'd like to avoid that if possible since we'd then need to create a FileProvider here https://github.com/aspnet/UserSecrets/blob/dev/src/dotnet-user-secrets/Program.cs#L227 and https://github.com/aspnet/UserSecrets/blob/dev/src/dotnet-user-secrets/Program.cs#L244